GameServerLib/API/ApiEventManager.cs

```diff
@@ -152,4 +153,36 @@ public void Publish(IChampion champion)
             });
         }
     }
+
+    public class EventOnHitUnit
+    {
+        private List<Tuple<object, IObjAiBase, Action<IAttackableUnit, bool>>> listeners = new List<Tuple<object, IObjAiBase, Action<IAttackableUnit, bool>>>();
+        public void AddListener(object owner, IObjAiBase unit, Action<IAttackableUnit, bool> callback)
+        {
+            var listenerTuple = new Tuple<object, IObjAiBase, Action<IAttackableUnit, bool>>(owner, unit, callback);
+            listeners.Add(listenerTuple);
+        }
+        public void RemoveListener(object owner, IObjAiBase unit)
+        {
+            listeners.RemoveAll((listener) => listener.Item1 == owner && listener.Item2 == unit);
+        }
+        public void RemoveListener(object owner)
+        {
+            listeners.RemoveAll((listener) => listener.Item1 == owner);
+        }
+        public void Publish(IObjAiBase unit, IAttackableUnit target, bool isCrit)
+        {
+            listeners.ForEach((listener) =>
+            {
+                if (listener.Item2 == unit)
```

Comment:
If you really do reference checking you need to use just HashTable or something similar and not List.

Reply:
Opened issue #796 .